src/Employer/Employer.Web/Orchestrators/Part1/LocationOrchestrator.cs

```diff
@@ -152,41 +179,86 @@ private async Task<LocationViewModel> GetViewModelAsync(Vacancy vacancy, bool? h
+            
             if (vacancy.EmployerLocation != null && (!hasLegalEntityChanged.HasValue || hasLegalEntityChanged == false))
             {
-                vm.UseOtherLocation = false;
 
-                if (vacancy.EmployerLocation.ToString() != legalEntity.Address.ToString())
+                if (vacancy.EmployerLocation.ToString() == legalEntity.Address.ToString())
+                    vm.Location = LegalEntityLocation;
+                else
                 {
-                    vm.UseOtherLocation = true;
-                    vm.AddressLine1 = vacancy.EmployerLocation.AddressLine1;
-                    vm.AddressLine2 = vacancy.EmployerLocation.AddressLine2;
-                    vm.AddressLine3 = vacancy.EmployerLocation.AddressLine3;
-                    vm.AddressLine4 = vacancy.EmployerLocation.AddressLine4;
-                    vm.Postcode = vacancy.EmployerLocation.Postcode;
+                    var employerLocation = vacancy.EmployerLocation.ToString();
+                    StringComparer comparer = StringComparer.OrdinalIgnoreCase;
+                    var otherLocations = vm.OtherLocationsAddress;
+                    if (otherLocations?.Any() == true)
+                    {
+                        foreach (var location in otherLocations)
+                        {
+                            if (comparer.Compare(employerLocation, location.ToString()) == 0)
+                            {
+                                vm.Location = location.ToString();
+                            }
+                        }
+                    }
                 }
-            }            
+            }
+
             return vm;
-        }        
+        }
 
-       private async Task UpdateEmployerProfileAsync(string employerAccountId, long legalEntityId, string tradingName, VacancyUser user)
+        private async Task UpdateEmployerProfileAsync(string employerAccountId,
+            VacancyEmployerInfoModel employerInfoModel, LocationEditModel locationEditModel, VacancyUser user)
         {
+            var legalEntityId = employerInfoModel.LegalEntityId.GetValueOrDefault();
+            var tradingName = employerInfoModel.NewTradingName;
+
             var employerProfile =
                 await _recruitVacancyClient.GetEmployerProfileAsync(employerAccountId, legalEntityId);
 
             if (employerProfile == null)
             {
-                throw new NullReferenceException($"No Employer Profile was found for employerAccount: {employerAccountId}, legalEntity: {legalEntityId}");
+                throw new NullReferenceException(
+                    $"No Employer Profile was found for employerAccount: {employerAccountId}, legalEntity: {legalEntityId}");
             }
 
-            if (employerProfile.TradingName != tradingName)
+            if (employerInfoModel.EmployerNameOption == EmployerNameOptionViewModel.NewTradingName &&
+                employerProfile.TradingName != tradingName)
```

Comment:
Do we need this extra logic `employerProfile.TradingName != tradingName`? The end result is the same if we remove this logic.  The `employerProfile.TradingName` needs to be equal to `employerInfoModel.NewTradingName`. So just set it if `EmployerNameOption` is `NewTradingName`.